fe/fe-core/src/main/java/com/starrocks/catalog/ResourceMgr.java

```diff
@@ -144,9 +144,11 @@ public ProcResult fetchResult() {
 
             for (Map.Entry<String, Resource> entry : nameToResource.entrySet()) {
                 Resource resource = entry.getValue();
-                // check resource privs
-                if (!Catalog.getCurrentCatalog().getAuth().checkResourcePriv(ConnectContext.get(), resource.getName(),
-                        PrivPredicate.SHOW)) {
+                // Since nameToResource.entrySet is not thread safe, resource may be dropped
```

Comment:
```suggestion
                // Since nameToResource.entrySet is not thread safe, resource may be dropped
```
IMO, nameToResource.entrySet() is thread safe, but its result will be changed after it is called.

Reply:
+1 @imay i will change the comments. what i really mean was
```
but its result will be changed after it is called.
```
thanks !